webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala

```diff
@@ -315,7 +315,9 @@ class CachedNodeChangesServiceImpl(
    * important than responsiveness.
    */
   def update(lowestId: Long, highestId: Long): Box[Unit] = {
-    addUpdate(ChangesUpdate.For(lowestId, highestId)).toBox
+    if (lowestId < highestId) {
```

Comment:
So, this one is interesting and might be a genuine independant bug. We weren't checking for the case when nothing happened (lowestId == highestId), and so we were alway adding an empty event, and doing a lot of work for nothing. 
I'm more at ease to do that in 8.2 because everything is complicated in that part of Rudder, but it *could* be backported in 8.1 since it's a trivial change to revert in the worst case.